core-blocks/latest-comments/index.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Server-side rendering of the `core/latest-comments` block.
+ *
+ * @package gutenberg
+ */
+
+// Used to get `_draft_or_post_title`, which prevents empty-title posts from
+// appearing with no title.
+require_once( ABSPATH . 'wp-admin/includes/template.php' );
+
+$default_comments_to_show = 5;
```

Comment:
As I understand it, scoping in PHP won't work as expected with what we have here. This wuld need to be accessed by `global`, which we're not doing.

Simpler demonstration:

_Before:_

```php
<?php

$extra = 'bar';

function do_foo() {
	echo 'foo' . $extra;
}

do_foo();
```

```
⇒ php test.php
foo%        
```

_After:_

```php
<?php

$extra = 'bar';

function do_foo() {
	global $extra;
	echo 'foo' . $extra;
}

do_foo();
```

```
⇒ php test.php
foobar%     
```

Though, as written, it's more of a constant, which is exemplified in [`define`](http://php.net/manual/en/function.define.php).

Though, above all, does this really need to be in the global scope? I think we're over-optimizing at the risk of outside breakage (anything else in the running WordPress page could access this variable, intentionally or not).

Reply:
Yeah I didn't think using `define` was right because I wanted a constant scoped to this file–and the constants created by `define` are global... right?

I'll admit that my PHP is wildly rusty and I'm totally unclear what the best practice is for this.

I just want a file-scoped constant. I can remove it if that's a pain to make.

Reply:
>and the constants created by `define` are global... right?

Yes, and so is the previous variable assignment, as can be demonstrated:

```diff
diff --git a/core-blocks/latest-comments/index.php b/core-blocks/latest-comments/index.php
index 33c21e41f..092fe08eb 100644
--- a/core-blocks/latest-comments/index.php
+++ b/core-blocks/latest-comments/index.php
@@ -9,6 +9,7 @@
 // appearing with no title.
 require_once( ABSPATH . 'wp-admin/includes/template.php' );
 
+$default_comments_to_show = 5;
 define( 'GUTENBERG_LATEST_COMMENTS_BLOCK_DEFAULT_TO_SHOW', 5 );
 
 /**
diff --git a/lib/load.php b/lib/load.php
index f1a8b09c6..f8a4b7985 100644
--- a/lib/load.php
+++ b/lib/load.php
@@ -34,3 +34,5 @@ require dirname( __FILE__ ) . '/register.php';
 foreach ( glob( dirname( __FILE__ ) . '/../core-blocks/*/index.php' ) as $block_logic ) {
 	require $block_logic;
 }
+
+var_export( $default_comments_to_show ); exit;
```

Reply:
Related: http://php.net/manual/en/language.variables.scope.php

Reply:
Yeah, I didn't realise that 😢

If the `define` call is too global I'm fine with just hard-coding it in. 🤷‍♂️

Reply:
Since it's prefixed enough it shouldn't be a huge concern.

Reply:
Alternatively, we could consider enhancing our JSON schema support to handle `minimum` and `maximum`, rely on the validation behavior, and assign the default only once in the block registration.

https://spacetelescope.github.io/understanding-json-schema/reference/numeric.html#range

In fact, since we use REST validation under the hood, I wonder if this is already possible.

Reply:
It's already supported:

https://github.com/WordPress/wordpress-develop/blob/8f95800d52c1736d651ae6e259f90ad4a0db2c3f/src/wp-includes/rest-api.php#L1169-L1209

Reply:
There's also `enum` to take care of the `align` valid options:

https://github.com/WordPress/wordpress-develop/blob/8f95800d52c1736d651ae6e259f90ad4a0db2c3f/src/wp-includes/rest-api.php#L1120-L1125

https://spacetelescope.github.io/understanding-json-schema/reference/generic.html#enumerated-values

Reply:
Huh, nice! Okay, I'll look at that.